tools/toolchains/mbed_toolchain.py

```diff
@@ -1393,3 +1393,20 @@ def report(self):
         to_ret['linker'] = {'flags': copy(self.flags['ld'])}
         to_ret.update(self.config.report)
         return to_ret
+
+def is_std_c_lib_supported_instead_small_c_lib(target, toolchain):
```

Comment:
I'm having trouble following the logic of this function. It feels like it is trying to do too much, and there appears to be the potential for some logical combinations to be missed. What you need is a function is_library_supported(libname), and is_small_lib_requested(). The latter might not even be a function since it is only the first two lines from below.

Reply:
Please review 0e9c83e15985570a1b18cea3e794d8f92c9d9ee1